### PR TITLE
317 feature/add scaling for map nodes

### PR DESCRIPTION
**Pull request by cchandram**
### Summary ###

Issue #317 

This pull request introduces a new "Scale By" feature that allows map circles to be dynamically scaled based on selected data fields (such as population or count of languages). The implementation includes UI updates, parameter handling, and a new scaling logic hook, as well as tests to ensure correct scaling behavior. 

Map circles can be scaled by population, count of languages (when mapping territories), and population attested as well as count of territories (when mapping languages).

**Changes:**

**1. Scale By Parameter and UI Integration**
- Added a new `scaleBy` parameter to `PageParams`, with corresponding updates to types, defaults, and URL handling. This allows users to select a field by which map circles are scaled.
- Introduced a new `ScaleBySelector` component in the side panel, letting users choose the scaling field from allowed options.

**2. Scaling Logic Implementation**
- Added `useScale` hook to compute the appropriate radius multiplier for each object based on the selected `scaleBy` field, with support for logarithmic and square-root scaling for certain fields.
- Updated `MapCircles` and `HoverableCircle` components to use the scaling logic, so circles are drawn with the correct radii according to the selected scale. 

### Testing ###
Added unit tests for the scaling logic to verify correct behaviour for different fields and edge cases:
- default multiplier when scaleBy is undefined or 'None'
- latitude mapping (-90..90) -> radius multiplier 2..10
- population mapping uses sqrt + log transform and handles missing values (non-renderable)

### Screenshots ###
lang-nav-map-circles-scaled-1
lang-nav-map-circles-scaled-2

"Scale By" dropdown when mapping territories:
lang-nav-map-circles-scaled-4

"Scale By" dropdown when mapping languages:
lang-nav-map-circles-scaled-6

### CODE REVIEW

File: src/features/transforms/scales/__tests__/useScale.test.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        const latMax = getBaseLanguageData('max', 'LatMax');
+        latMax.latitude = 90;
+
+        const latMin = getBaseLanguageData('min', 'LatMin');
+        latMin.latitude = -90;
+
+        const { result } = renderHook(() => useScale({ objects: [latMax, latMin], scaleBy: SortBy.Latitude }));
+
+        // Max latitude should map to the maximum multiplier (10)
+        expect(result.current.getScale(latMax)).toBeCloseTo(10);
```
Comment (conradarcturus) [inline src/features/transforms/scales/__tests__/useScale.test.tsx]: oh! I didn't know about `toBeCloseTo` -- that's good because of float errors sometimes :p

File: src/features/transforms/scales/useScale.tsx [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+            if (numericValue <= 0) return 0;
+            if (shouldUseLogScale) return Math.log10(numericValue) / range;
+            return numericValue / range;
+        },
+        [scaleBy, tMin, tMax, range, shouldUseLogScale],
+    );
+
+    const getScale = useCallback(
+        (object: ObjectData) => {
+            if (!scaleBy || scaleBy === 'None') return 2; // default radius multiplier
```
Comment (conradarcturus) [inline src/features/transforms/scales/useScale.tsx]: leave this as 2

File: src/features/transforms/scales/useScale.tsx [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+        },
+        [scaleBy, getNormalizedValue],
+    );
+
+    return { scaleBy, getScale, maxValue, minValue };
+};
+
+export default useScale;
+
+function getMinimumValue(scaleBy?: ScaleBy): number {
```
Comment (conradarcturus) [inline src/features/transforms/scales/useScale.tsx]: I'd move the functions you copied from `useColor` and just put it in the `@features/transforms` directory (the parent one to this) make a separate `getMinimumValue.ts` and `getMaximumValue.ts` (or maybe put them in the same file?)

But how do you handle both a ScaleBy and a ColorBy? very optimistically think you can do the suggestions below.
```suggestion
export function getMinimumValue(field?: ColorBy | ScaleBy): number {
```

File: src/features/transforms/coloring/ColorTypes.ts [hunk 4]
```diff
@@ -1,7 +1,57 @@
...
+  ];
+
+  let filteredColorBys = [...allowedColorBys];
+
+  // Exclude area and count-of-languages when mapping Languages (use CountOfDialects instead)
+  if (objectType === ObjectType.Language) {
+    filteredColorBys = filteredColorBys.filter(
+      (s) => s !== SortBy.CountOfLanguages && s !== SortBy.Area,
+    );
+  }
```
[the remark this answers is not in the document]
Comment (conradarcturus) [reply src/features/transforms/coloring/ColorTypes.ts]: I made a task to follow up on this -- let me know if you are interested in it or I can just leave it unassigned. https://github.com/Translation-Commons/lang-nav/issues/333
